// compass-brunch/lib/index.js
// Generated by CoffeeScript 1.3.3
(function() {
  var CompassCompiler, child_process, fs, sysPath;

  child_process = require('child_process');

  fs = require('fs');

  sysPath = require('path');

  module.exports = CompassCompiler = (function() {

    CompassCompiler.prototype.brunchPlugin = true;

    CompassCompiler.prototype.type = 'stylesheet';

    CompassCompiler.prototype.extension = 'scss';

    CompassCompiler.prototype.pattern = /\.s[ac]ss$/;

    function CompassCompiler(config) {
      var configPath,
        _this = this;
      this.config = config;
      child_process.exec("compass --version", function(error, stdout, stderr) {
        if (error) {
          console.error("You need to have Compass on your system");
          return console.error("Execute `gem install compass`");
        }
      });
      fs.exists(this.config.paths.compass, function(exists) {
        if (!exists) {
          console.error("Compass config file doesn't exist");
          return console.error(_this.config.paths.compass);
        }
      });
      configPath = sysPath.join(process.cwd(), this.config.paths.compass);
      child_process.exec("compass compile --config " + configPath, function(error, stdout, stderr) {
        if (error !== null) {
          return console.log("exec error: " + error);
        }
      });
    }

    return CompassCompiler;

  })();

}).call(this);
